compass/sign/RemoteSignatureSource.java

```diff
@@ -40,15 +48,66 @@ public RemoteSignatureSource(String uri,
                                String trustCertCollectionFilePath,
                                String clientCertChainFilePath,
                                String clientPrivateKeyFilePath) throws SSLException {
-    this(NettyChannelBuilder
+    this(OkHttpChannelBuilder
```

Comment:
Question:
So just using `OkHttpChannelBuilder` will magically make our problems go away?

I suppose this is the reason for the solution:
https://github.com/grpc/grpc-java/blob/master/core/src/main/java/io/grpc/internal/DnsNameResolver.java#L113

Reply:
So it seems https://github.com/grpc/grpc-java/issues/1463#issuecomment-346283010 .

It is a bit of a complicated issue for what I read, but the code reference you are pointing to only applies to connection failures re-attempts rather than the connection going idle. Our code currently does not tolerate a connection failure, it would throw an exception up the stack. What we want is for the channel builder to re-evaluate the DNS record each time the connection goes stale == idle, which is a timeout currently set on 5 seconds.